packages/testing/src/test/chai-retry-plugin.unit.ts

```diff
@@ -0,0 +1,290 @@
+        });
+
+        it('.empty', async () => {
+            const { funcToRetry, getAttempts } = getFunctionToRetry((attempts) => (attempts !== 5 ? [1, 2, 3] : []));
+
+            await expect(funcToRetry).retry().to.be.empty;
+
+            expect(getAttempts()).to.equal(5);
+        });
+
+        it('.property(), .true', async () => {
+            const { funcToRetry, getAttempts } = getFunctionToRetry((attempts) => {
+                if (attempts < 3) {
+                    throw new Error('Failed');
+                }
+                return { success: true };
+            });
+
+            await expect(funcToRetry).retry().and.have.property('success').and.be.true;
+
+            expect(getAttempts()).to.equal(3);
+        });
+
+        it('.sealed', async () => {
+            const sealedObject = Object.seal({ prop1: 'value1', prop2: 'value2' });
+
+            await expect(() => sealedObject).retry().to.be.sealed;
+        });
+
+        it('.not.sealed', async () => {
+            const notSealedObject = { prop1: 'value1', prop2: 'value2' };
+
+            try {
+                await expect(() => notSealedObject).retry({ retries: 3 }).to.be.sealed;
+            } catch (error: unknown) {
+                expect((error as Error).message).to.include('3 retries');
+            }
+        });
+    });
+
+    describe('should work with various assertion methods:', () => {
+        it('.satisfy()', async () => {
+            const { funcToRetry, getAttempts } = getFunctionToRetry((attempts) => ({
+                attempts,
+            }));
+
+            await expect(funcToRetry)
+                .retry()
+                .to.satisfy((obj: { attempts: number }) => obj.attempts > 3);
+
+            expect(getAttempts()).to.equal(4);
+        });
+
+        it('.have.property(), and.be.above()', async () => {
+            const { funcToRetry, getAttempts } = getFunctionToRetry((attempts) =>
+                attempts < 4 ? { notValue: 2 } : { value: attempts }
+            );
+
+            await expect(funcToRetry).retry().have.property('value').and.be.above(4);
+
+            expect(getAttempts()).to.equal(5);
+        });
+
+        it('.to.increase(), .by()', async () => {
+            let attempts = 0;
+            const myObj = { val: 1 };
+
+            const addTwo = () => {
+                attempts++;
+                myObj.val += 2;
+            };
+
+            await expect(addTwo).retry().to.increase(myObj, 'val').by(2);
+
+            expect(attempts).to.equal(1);
+        });
+
+        it('.oneOf()', async () => {
+            const { funcToRetry, getAttempts } = getFunctionToRetry((attempts) => attempts);
+
+            await expect(funcToRetry).retry().to.not.be.oneOf([1, 2, 3]);
+            expect(getAttempts()).to.equal(4);
+        });
+
+        it('.change()', async () => {
+            let attempts = 0;
+            const myObj = { dots: '', comas: '' };
+
+            const addDot = () => {
```

Comment:
you wrote a helper for the `attempts` stuff.. use it

Reply:
@yurii-ve fixed